### PR TITLE
Use Docker volumes for storing sites

### PR DESCRIPTION
**Pull request by tbolender**
When setting up ERPNext in our environment, I stumbled upon the usage of the `sites` folder as volume. This can cause problems regarding permissions depending on your operating system and/or configuration. I received permission errors preventing the creation of a new site.

Especially in production setup, this is not a nice solution. I propose the usage of volumes for the site storage as well. If really required, access to the folder is (at least under Linux) still possible.

P.S. Thanks a lot for your work, it made setting up of a page pretty straight forward!

### CODE REVIEW

**Comment by revant** [comment]
>  I received permission errors preventing the creation of a new site.
If the sites directory is owned by uid 1000 and gid 1000, generally first user on linux then there won't be any permission problem

I don't mind it being Docker volume instead of bind mount. It is easy for users to edit sites config directly from host if it's bind mount.

Additional best practice has to be explained to "docker run" new container with shared volume to edit config
(AWS rds needs change to config json). This should also be part of documentation then?

Making it docker volume is first step to multi host volumes

**Comment by tbolender** [comment]
@ddavidebor Updated. I also fixed all backup and restore instructions. I missed it before.

@revant I am not sure whether I correctly get what for instructions you mean. To launch a one-time container to edit the config in the console? Which image should we use for this?

**Comment by revant** [comment]
> To launch a one-time container to edit the config in the console? Which image should we use for this?

check this issue: https://github.com/frappe/frappe_docker/issues/183

user has to add `"rds_db": 1` to `common_site_config.json`. For that you need to access the file.

in case of bind mount it is as easy as editing the file from host.

in case of docker volume, another container with editor has to run with sites-vol mounted to edit common_site_config.json

File: README.md [hunk 1]
```diff
@@ -250,7 +250,7 @@ Environment Variables
     -e "SECRET_ACCESS_KEY=secret_access_from_provider" \
     -e "ENDPOINT_URL=https://region.storage-provider.com" \
     -e "BUCKET_DIR=frappe-bench-v12" \
-    -v ./installation/sites:/home/frappe/frappe-bench/sites \
+    -v sites-vol:/home/frappe/frappe-bench/sites \
```
Comment (revant) [inline README.md:253]:
```suggestion
    -v <project-name>_sites-vol:/home/frappe/frappe-bench/sites \
```

volumes generally have prefix when created as part of docker-compose

**Comment by revant** [comment]
How do I setup AWS RDS in this setup? I need to edit `sites/common_site_config.json` for that

https://github.com/frappe/frappe/wiki/Using-Frappe-with-Amazon-RDS-(or-any-other-DBaaS)

Common question in Issues and on forum.

File: README.md (hunk 1 above)
Comment (tbolender) [reply README.md]: Fixed in one other place as well.

**Pull request merged**